src/Generics/SOP/TH.hs

```diff
@@ -130,7 +130,11 @@ deriveMetadataValue n codeName datatypeInfoName = do
              , funD datatypeInfoName' [clause [] (normalB $ metadata' isNewtype name cons) []] -- treeDatatypeInfo = ...
              ]
 
+#if MIN_VERSION_template_haskell(2,11,0)
+deriveGenericForDataDec :: Bool -> Cxt -> Name -> [TyVarBndr] -> [Con] -> Cxt -> Q [Dec]
```

Comment:
the `Cxt` / `[Name]` parameter isn't used anywhere; I'm not sure what would be wanted approach. This works for now.